src/test/java/com/example/project/test/ShoppingListTest.java

```diff
@@ -0,0 +1,138 @@
+        driver = new AndroidDriver<>(url, cap);
+
+        driver.manage().timeouts().implicitlyWait(15, TimeUnit.SECONDS);
+    }
+
+    @BeforeMethod
+    public void resetApp() {
+        driver.resetApp();
+        driver.findElementById("com.android.permissioncontroller:id/continue_button").click();
+        driver.findElementById("android:id/button1").click();
+    }
+
+    @Test
+    public void createNewList_successful() throws InterruptedException {
+        final String listName = "New list";
+        AllListsPage allListsPage = createNewList(listName);
+        Thread.sleep(2000);
+        driver.hideKeyboard();
+        driver.navigate().back();
+        assertThat(allListsPage.firstListNameTextView.getText()).isEqualTo(listName);
+    }
+
+    @Test
+    public void addItemToList_successful() {
+        createNewList("New list");
+
+        waitForListPageToLoad();
+
+        ListPage listPage = addItemToList("Milk", "1.2", "2", "4", "2");
+
+        assertThat(listPage.firstItemNameTextView.getText()).isEqualTo("Milk");
+    }
+
+    @Test
+    public void editItem() {
+        createNewList("New list");
+
+        waitForListPageToLoad();
+
+        ListPage listPage = addItemToList("Milk", "1.2", "2", "4", "2")
+                .longPressOnFirstItem()
+                .clickOnEdit()
+                .enterComment("3% fat")
+                .enterPrice("1.3")
+                .saveItem();
+
+        assertThat(listPage.firstItemPriceTextView.getText()).isEqualTo("1.3 £");
+        assertThat(listPage.firstItemCommentTextView.getText()).isEqualTo("3% fat");
+    }
+
+    @Test
+    public void removeItem() {
+        createNewList("New list");
+
+        waitForListPageToLoad();
+
+        addItemToList("Milk", "1.2", "2", "4", "2")
+                .longPressOnFirstItem()
+                .clickOnRemove().
+                confirmRemoval();
+
+        assertThatExceptionOfType(NoSuchElementException.class).isThrownBy(() -> driver.findElementByXPath("//android.widget.ListView/android.widget.RelativeLayout"));
+    }
+
+    @Test
+    public void checkTotal() {
+        AllListsPage allListsPage = createNewList("New list");
+
+        waitForListPageToLoad();
+
+        addItemToList("Milk", "1.2", "2", "4", "2");
+        addItemToList("Butter", "3", "1", "1", "2");
+
+        driver.hideKeyboard();
+        driver.navigate().back();
+
+        assertThat(allListsPage.firstListInformationTextView.getText()).contains("Sum: 5.4 £");
+    }
+
+    private AllListsPage createNewList(String listName) {
+        return new AllListsPage(driver).enterNewListName(listName).clickCreateNewListButton();
+    }
+
+    private void waitForListPageToLoad() {
+        WebDriverWait wait = new WebDriverWait(driver, 20);
+        wait.until(ExpectedConditions.visibilityOfElementLocated(By.id("com.slava.buylist:id/editText1")));
+    }
+
+    private ListPage addItemToList(String name, String price, String amount, String unit, String category) {
```

Comment:
shouldn't this and above methods be located in the `ListPage` or other `Page` classes?